### DISCUSSION TITLE
How to enable CPUManagerPolicy static options

### DISCUSSION THREAD

**Discussion by schlichtanders** [mysticaltech/terraform-hcloud-kube-hetzner#833] [Q&A]
I think I have the problem that my pods don't get dedicated CPUs.
Researching pointed me to https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#static-policy-options

which mentions to set `full-pcpus-only` for the CPUManager Policy, which I think is set via `kubelet-arg`

I've seen that this repo mentions `kubelet-arg` in locals.tf, however I don't know how I can set this from my `kube.tf` file. Any help is highly appreciated.

**Comment by schlichtanders** [comment]
I just found the option `k3s_exec_server_args` - is it safe to add `kubelet-arg` here?

**Reply by schlichtanders** [reply]
indeed it seems that `--kubelet-arg` is normally used several times, so there shouldn't be any conflict (I found [this example usage](https://github.com/k3s-io/k3s/issues/5488#issuecomment-1212693825) somewhere on github)

**Comment by mysticaltech** [comment]
@schlichtanders Sadly, I do not think it will work, please do not hesitate to submit a PR, what needs to be done is add a variable and concatenate the value to the value in this line:
 
  https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/7a731ebbc81e60b11324d5db18b09589d633306b/locals.tf#L322

**Reply by schlichtanders** [reply]
what is the reason why `k3s_exec_server_args` would not work?

**Reply by schlichtanders** [reply]
according to the [k3s server docs](https://docs.k3s.io/installation/configuration) the server args take precedence. Hence it should work if the original kubelet-args are included. I will try it out

**Reply by mysticaltech** [reply]
@schlichtanders In which case, make sure to include the params listed above on line 322 of the locals.tf file, as those are needed.
